### PR TITLE
Documentation for Real Time Collaboration

### LINKED ISSUE #10227: Docs: Update docs on real time collaboration
From https://github.com/jupyterlab/jupyterlab/pull/10118

---

I think we should document the collaborative "shared model" approach in https://jupyterlab.readthedocs.io/en/stable/. The patterns described in the jupyterlab/rtc repository seem to be out-of-date now.

1. The `shared-models` package contains a `SharedNotebook` and a `SharedFile` which are the abstract interfaces to work against if you want to manipulate a notebook or a file. In a MVVM pattern, the `shared-models` are the model. The view is implemented by the other packages (`codemirror`, `cells`, `notebook`, ..) which render the shared model. The model enables you to manipulate and observe data using a well-defined interface - completely independent from the networking technology that is used to synchronize the model. 
2. A useful resource: `jlpm docs` generates documentation for the collaborative interfaces exposed by the `shared-models` package.
2. I imagine that there is already a lot of documentation on how to manipulate notebooks using `notebook/model` or `cell/model` or `codeeditor/model`. Now everything is bundled into a single interface. We should work towards replacing the existing documentation to describe the new `SharedNotebook` interface instead. Eventually, we should deprecate the old model implementations in favor of a single consistent & collaborative interface.
2. We should document how to add more shared-models to Jupyter. It is fairly easy to define new models (even outside of the `shared-models` package). @hbcarlos is already working on a shared model for Draw.io. The "shared model" is implemented using Yjs https://docs.yjs.dev/ and exposes a Yjs Document that the provider syncs to other devices. 
3. The new `docprovider` package is one implementation that enables peers to synchronize shared-models / Yjs documents over a websocket-python server. It is possible to implement different providers (e.g. based on any existing Yjs provider) or synchronize the shared models to any service provider that can synchronize Yjs documents.

Regarding 1.: The pattern that we established is to create an abstract interface around the "thing" that we want to make collaborative. Internally, we use Yjs to implement that model. Then we use Yjs' providers to synchronize the underlying Yjs document to other peers. These are patterns that are already established in the general Yjs ecosystem. So it might make sense to link to https://docs.yjs.dev/ at some point.

This is just a short description of the inner workings and the patterns that we want to establish. Please let me know if you want to discuss anything in more detail.

Furthermore, we should probably clarify in `jupyterlab/rtc` that we arrived at a single solution. Further work should be coordinated directly on the `jupyterlab/jupyterlab` repository.

### PR DESCRIPTION
**Pull request by hbcarlos**
Documentation for Real Time Collaboration.

Also updated other parts of the documentation like [JupyterLab URLs](https://jupyterlab.readthedocs.io/en/stable/user/urls.html#jupyterlab-urls):
* I think JupyterLab doesn't start in single-mode by default
* Now we can open a Notebook in two tabs simultaneously.

## References
Fixes #10227

### CODE REVIEW

**Comment by krassowski** [comment]
How users should access the link to share the session? Should they install the `jupyterlab-link-share` extension? Should it be moved into the core? See https://github.com/jupyterlab/jupyterlab/issues/5382#issuecomment-873979368 (I think it is a common question).

**Comment by hbcarlos** [comment]
Thanks for pointing that out, @krassowski. I completely forgot to mention it in the documentation.

There is a "Copy Shareable Link" entry on the context menu. I'll add to the documentation.
share-link

**Comment by krassowski** [comment]
Maybe the easiest approach would be to add the token to it?

**Comment by hbcarlos** [comment]
Yes, I'll open an issue to track this and the corresponding PR.

File: docs/source/extension/documents.rst [hunk 1]
```diff
@@ -104,6 +135,24 @@ the Document Manager. Contexts are not meant to be subclassed or
...
 
+Shared Models
+-------------
+
+The shared-models package contains an `ISharedNotebook
+<../api/interfaces/shared_models.isharednotebook.html>`_ and an `ISharedFile
+<../api/interfaces/shared_models.isharedfile.html>`_ which are the abstract
+interfaces to work against if you want to manipulate a notebook or a file.
+
+These models wrap a `Yjs documents (Y.Doc) <https://docs.yjs.dev/api/y.doc>`_ which represents
```
Comment (blink1073) [inline docs/source/extension/documents.rst:146]:
```suggestion
These models wrap a `Yjs document (Y.Doc) <https://docs.yjs.dev/api/y.doc>`_ which represents
```

File: docs/source/user/rtc.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Moreover, you can see the cursors from other users with an anonymous
+username, a username that will disappear in a few seconds to make room
+for what is essential, the document's content.
+
+.. image:: images/shared_cursors.png
+   :align: center
+   :class: jp-screenshot
+
+Something you need to be aware of is that not all editors are in sync.
```
Comment (blink1073) [inline docs/source/user/rtc.rst:29]:
```suggestion
Something you need to be aware of is that not all editors in JupyterLab support sync.
Additionally, opening the same underlying document using different editor types may result
in a lack of synchronization.
```

**Review by blink1073 — APPROVED** [review]
Thanks!  Just a couple suggestions.

**Comment by hbcarlos** [comment]
Thanks, Steven!

The link check test fails because I added a new file, `rtc.rst`, so the "Edit on GitHub" link from Read the Docs doesn't work.

**Pull request merged**